pkg/server/conn.go

```diff
@@ -253,6 +253,13 @@ func (cc *clientConn) authSwitchRequest(ctx context.Context, plugin string) ([]b
 		clientPlugin += "_client"
 	} else if plugin == mysql.AuthLDAPSimple {
 		clientPlugin = mysql.AuthMySQLClearPassword
+	} else if authPluginImpl, ok := cc.extensions.GetAuthPlugins()[plugin]; ok {
+		if authPluginImpl.RequiredClientSidePlugin != "" {
+			clientPlugin = authPluginImpl.RequiredClientSidePlugin
+		} else {
+			// If the plugin does not need special handling of the password, use clear text password.
+			clientPlugin = mysql.AuthMySQLClearPassword
```

Comment:
Seems this behavior is different with the comment of `RequiredClientSidePlugin`:

> If this is empty then `AuthPlugin.Name` is used as the required client-side plugin.